### ISSUE TITLE
[BUG] Impossible to connect to redis using a unix socket

### ISSUE THREAD

**Issue by amo13** [paperless-ngx/paperless-ngx#2072]
### Description

On my setup, paperless connects to redis using a unix socket. Since 1.10.0 this does not work any more. The cause seems to be the format of the PAPERLESS_REDIS variable in the `paperless.conf` config file.
If I want celery to connect to redis successfully using a socket, I [need to use this format](https://docs.celeryq.dev/en/stable/getting-started/backends-and-brokers/redis.html): `PAPERLESS_REDIS='redis+socket:///run/redis/redis.sock'`. This is the only way to get celery to connect successfully.
Unfortunately, the rest of paperless does not seem to understand this format and gives the following error:
```
Nov 30 18:34:48 test celery[17178]: [2022-11-30 18:34:48,696] [INFO] [celery.worker.strategy] Task documents.tasks.consume_file[40c8ef18-4be7-4a1f-98d9-f1da403ac1f2] received
Nov 30 18:34:48 test celery[17215]: [2022-11-30 18:34:48,897] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[40c8ef18-4be7-4a1f-98d9-f1da403ac1f2] raised unexpected: ValueError('Redis URL must specify one of the following schemes (redis://, rediss://, unix://)')
Nov 30 18:34:48 test celery[17215]: Traceback (most recent call last):
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task
Nov 30 18:34:48 test celery[17215]:     R = retval = fun(*args, **kwargs)
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__
Nov 30 18:34:48 test celery[17215]:     return self.run(*args, **kwargs)
Nov 30 18:34:48 test celery[17215]:   File "/usr/share/paperless/src/documents/tasks.py", line 177, in consume_file
Nov 30 18:34:48 test celery[17215]:     document = Consumer().try_consume_file(
Nov 30 18:34:48 test celery[17215]:   File "/usr/share/paperless/src/documents/consumer.py", line 272, in try_consume_file
Nov 30 18:34:48 test celery[17215]:     self._send_progress(0, 100, "STARTING", MESSAGE_NEW_FILE)
Nov 30 18:34:48 test celery[17215]:   File "/usr/share/paperless/src/documents/consumer.py", line 77, in _send_progress
Nov 30 18:34:48 test celery[17215]:     async_to_sync(self.channel_layer.group_send)(
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/asgiref/sync.py", line 218, in __call__
Nov 30 18:34:48 test celery[17215]:     return call_result.result()
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/concurrent/futures/_base.py", line 451, in result
Nov 30 18:34:48 test celery[17215]:     return self.__get_result()
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/concurrent/futures/_base.py", line 403, in __get_result
Nov 30 18:34:48 test celery[17215]:     raise self._exception
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/asgiref/sync.py", line 284, in main_wrap
Nov 30 18:34:48 test celery[17215]:     result = await self.awaitable(*args, **kwargs)
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/channels_redis/core.py", line 546, in group_send
Nov 30 18:34:48 test celery[17215]:     connection = self.connection(self.consistent_hash(group))
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/channels_redis/core.py", line 738, in connection
Nov 30 18:34:48 test celery[17215]:     self.pools[index] = self.create_pool(index)
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/channels_redis/core.py", line 133, in create_pool
Nov 30 18:34:48 test celery[17215]:     return aioredis.ConnectionPool.from_url(host["address"])
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/redis/asyncio/connection.py", line 1409, in from_url
Nov 30 18:34:48 test celery[17215]:     url_options = parse_url(url)
Nov 30 18:34:48 test celery[17215]:   File "/usr/lib/python3.10/site-packages/redis/asyncio/connection.py", line 1345, in parse_url
Nov 30 18:34:48 test celery[17215]:     raise ValueError(
Nov 30 18:34:48 test celery[17215]: ValueError: Redis URL must specify one of the following schemes (redis://, rediss://, unix://)
```

### Steps to reproduce

1. Set redis to listen to socket
2. Configure paperless to connect to redis using the socket
3. Import a file
4. Does not work. 

### Webserver logs

```bash
nothing
```

### Paperless-ngx version

1.10.0

### Host OS

Archlinux

### Installation method

Bare metal

### Browser

Firefox

### Configuration changes

PAPERLESS_REDIS='redis+socket:///run/redis/redis.sock'

### Other

Would be awesome if paperless would also support using the [virtual_host URL parameter](https://docs.celeryq.dev/en/stable/getting-started/backends-and-brokers/redis.html) when connecting to redis using a unix socket.

**Comment by stumpylog** [comment]
For a Unix socket, the channels documentation says it should be unix:///path/to/redis.sock.  So that's the compatibility issue

**Comment by amo13** [comment]
Seems to be working fine! Did not test with docker though, since I do not use docker. I just manually applied your commit in the feature-socket-compat-layer branch. Anyway, the error is gone, thank you! Looking forward to have this in the next release.

--------

Don't know if that's related, but I see alot of `event loop is closed` errors in the journal logs of celery:

```
Dec 02 22:28:31 test celery[1866]: [2022-12-02 22:28:31,609] [ERROR] [asyncio] Task exception was never retrieved
Dec 02 22:28:31 test celery[1866]: future: <Task finished name='Task-17' coro=<Connection.disconnect() done, defined at /usr/lib/python3.10/site-packages/redis/asyncio/connection.py:828> ex>
Dec 02 22:28:31 test celery[1866]: Traceback (most recent call last):
Dec 02 22:28:31 test celery[1866]:   File "/usr/lib/python3.10/site-packages/redis/asyncio/connection.py", line 837, in disconnect
Dec 02 22:28:31 test celery[1866]:     self._writer.close()  # type: ignore[union-attr]
Dec 02 22:28:31 test celery[1866]:   File "/usr/lib/python3.10/asyncio/streams.py", line 337, in close
Dec 02 22:28:31 test celery[1866]:     return self._transport.close()
Dec 02 22:28:31 test celery[1866]:   File "/usr/lib/python3.10/asyncio/selector_events.py", line 698, in close
Dec 02 22:28:31 test celery[1866]:     self._loop.call_soon(self._call_connection_lost, None)
Dec 02 22:28:31 test celery[1866]:   File "/usr/lib/python3.10/asyncio/base_events.py", line 753, in call_soon
Dec 02 22:28:31 test celery[1866]:     self._check_closed()
Dec 02 22:28:31 test celery[1866]:   File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
Dec 02 22:28:31 test celery[1866]:     raise RuntimeError('Event loop is closed')
Dec 02 22:28:31 test celery[1866]: RuntimeError: Event loop is closed
Dec 02 22:28:32 test celery[1866]: [2022-12-02 22:28:32,324] [INFO] [paperless.consumer] Document 2022-04-19 Linde Provinzial KFZ Versicherung 12-2021 consumption finished
Dec 02 22:28:32 test celery[1866]: [2022-12-02 22:28:32,327] [ERROR] [asyncio] Task exception was never retrieved
Dec 02 22:28:32 test celery[1866]: future: <Task finished name='Task-21' coro=<Connection.disconnect() done, defined at /usr/lib/python3.10/site-packages/redis/asyncio/connection.py:828> ex>
Dec 02 22:28:32 test celery[1866]: Traceback (most recent call last):
Dec 02 22:28:32 test celery[1866]:   File "/usr/lib/python3.10/site-packages/redis/asyncio/connection.py", line 837, in disconnect
Dec 02 22:28:32 test celery[1866]:     self._writer.close()  # type: ignore[union-attr]
Dec 02 22:28:32 test celery[1866]:   File "/usr/lib/python3.10/asyncio/streams.py", line 337, in close
Dec 02 22:28:32 test celery[1866]:     return self._transport.close()
Dec 02 22:28:32 test celery[1866]:   File "/usr/lib/python3.10/asyncio/selector_events.py", line 698, in close
Dec 02 22:28:32 test celery[1866]:     self._loop.call_soon(self._call_connection_lost, None)
Dec 02 22:28:32 test celery[1866]:   File "/usr/lib/python3.10/asyncio/base_events.py", line 753, in call_soon
Dec 02 22:28:32 test celery[1866]:     self._check_closed()
Dec 02 22:28:32 test celery[1866]:   File "/usr/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
Dec 02 22:28:32 test celery[1866]:     raise RuntimeError('Event loop is closed')
Dec 02 22:28:32 test celery[1866]: RuntimeError: Event loop is closed
Dec 02 22:28:32 test celery[1866]: [2022-12-02 22:28:32,387] [INFO] [celery.app.trace] Task documents.tasks.consume_file[72fe2bda-d0e1-4546-86e8-ff8044d681e8] succeeded in 7.3881921439
```

**Comment by stumpylog** [comment]
Oh, missed the install method.  Thanks for testing.

For the other, double check your version of `channels-redis` is at 3.4.1.  That looks like a bug I reported to them on their 4.0 release that isn't fixed yet.

**Comment by eomanis** [comment]
Fixed for me with [paperless-ngx 1.10.2](https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.10.2), thank you.
